addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/discovery/internal/ModbusThingHandlerDiscoveryService.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.discovery.internal;
+
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+
+/**
+ * Implementation of this interface is responsible for discovery over
+ * a Modbus endpoint. Each time a supporting endpoint handler is created
+ * an instance of this service will be created as well and attached to the
+ * thing handler.
+ *
+ * @author Nagy Attila Gabor - initial contribution
+ */
+public interface ModbusThingHandlerDiscoveryService extends ThingHandlerService {
```

Comment:
[Should have @NonNullByDefault](https://www.openhab.org/docs/developer/development/conventions.html#null-annotations)?

Reply:
yes, it was required to conform with the rest of the package